### PR TITLE
Update testsuite to 646b43e

### PR DESCRIPTION
**Pull request by aardappel**
(this is HEAD~1 as of this writing, HEAD containing the Memory64 tests which I want to land separately)

### CODE REVIEW

File: test/regress/regress-35.txt [hunk 1]
```diff
@@ -20,7 +20,3 @@
   )
   (export "run" (func $run))
 )
-(;; STDERR ;;;
```
Comment (sbc100) [inline test/regress/regress-35.txt:23]: Do you know what happened here?

File: test/dump/relocations-section-target.txt [hunk 2]
```diff
@@ -39,5 +39,4 @@ Code Disassembly:
  000029: 10 80 80 80 80 00          | call 0 <env.b>
            00002a: R_WASM_FUNCTION_INDEX_LEB 0 <env.b>
  00002f: 0b                         | end
-
```
Comment (sbc100) [inline test/dump/relocations-section-target.txt:42]: What here?

**Review by sbc100 — COMMENTED** [review]
Otherwise lgtm

**Review by sbc100 — APPROVED** [review]

File: test/regress/regress-35.txt (hunk 1 above)
Comment (aardappel) [reply test/regress/regress-35.txt:23]: nope, result of `-r` on the tests

File: test/dump/relocations-section-target.txt (hunk 2 above)
Comment (aardappel) [reply test/dump/relocations-section-target.txt:42]: same

**Pull request merged**